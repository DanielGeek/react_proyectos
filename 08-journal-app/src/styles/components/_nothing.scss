.nothing__main-content {
    align-items: center;
    background-color: $primary;
    color: white;
    display: flex;
    flex-direction: columns;
    justify-content: center;
    height: 100vh;

    p {
        text-align: center;
    }
}