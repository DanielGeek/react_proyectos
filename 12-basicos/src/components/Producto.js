import React from 'react';

const Producto = ({ producto, productos, carrito, agregarProducto }) => {

    const { id, nombre, precio } = producto;

    const seleccionarProducto = id => {
        // busca el producto con el id pasado y retorna un nuevo array con ese obj
        // id[0] para retornar el obj
        const producto = productos.filter(producto => producto.id === id)[0];
        console.log(producto);
    }
    return (
        <div>
            <h2>{nombre}</h2>
            <p>${precio}</p>
            <button
                type="button"
                onClick={() => seleccionarProducto(id)}
            >Comprar</button>
        </div>);
}
export default Producto;