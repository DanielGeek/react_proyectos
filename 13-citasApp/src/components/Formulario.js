import React, { useState } from 'react';
import uuid from 'uuid/v4';

const Formulario = ({ crearCita }) => {

    // Crear State de Citas
    const [cita, actualizarCita] = useState({
        mascota: '',
        propietario: '',
        fecha: '',
        hora: '',
        sintomas: ''
    });

    const [error, actualizarError] = useState(false);

    const actualizarState = e => {
        // hago una copia del state actual y lo actualizo con el valor que corresponda
        actualizarCita({
            ...cita,
            [e.target.name]: e.target.value
        })
    }

    // Extraer los valores para asinarlos al value de cada input
    const { mascota, propietario, fecha, hora, sintomas } = cita;

    // Cuando el usuario presiona agregar cita
    const submitCita = e => {
        e.preventDefault();

        // Validar
        if (mascota.trim() === '' || propietario.trim() === '' || fecha.trim() === '' || hora.trim() === '' || sintomas.trim() === '') {
            actualizarError(true);
            return;
        }
        // ELiminar el mensaje previo
        actualizarError(false);

        // Asignar un ID
        cita.id = uuid();

        // Crear la cita
        crearCita(cita);

        // Reiniciar el form
    }

    return (
        <>
            <h2>Crear Cita</h2>
            { error ? <p className="alerta-error">Todos los campos son obligatorios</p>
                : null}
            <form
                onSubmit={submitCita}
            >
                <label>Nombre Mascota</label>
                <input
                    type="text"
                    name="mascota"
                    className="u-full-width"
                    placeholder="Nombre Mascota"
                    onChange={actualizarState}
                    value={mascota}
                />
                <label>Nombre Dueño</label>
                <input
                    type="text"
                    name="propietario"
                    className="u-full-width"
                    placeholder="Nombre Dueño de la Mascota"
                    onChange={actualizarState}
                    value={propietario}
                />

                <label>Fecha</label>
                <input
                    type="date"
                    name="fecha"
                    className="u-full-width"
                    onChange={actualizarState}
                    value={fecha}
                />
                <label>Hora</label>
                <input
                    type="time"
                    name="hora"
                    className="u-full-width"
                    onChange={actualizarState}
                    value={hora}
                />
                <label>Síntomas</label>
                <textarea
                    className="u-full-width"
                    name="sintomas"
                    onChange={actualizarState}
                    value={sintomas}
                ></textarea>
                <button
                    type="submit"
                    className="u-full-width button-primary"
                >Agregar Cita</button>
            </form>
        </>
    );
}

export default Formulario;