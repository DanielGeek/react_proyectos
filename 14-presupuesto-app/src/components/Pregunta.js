import React, { useState } from 'react';
import Error from './Error';

const Pregunta = () => {

    // definir el state
    const [cantidad, guardarCantidad] = useState(0);
    const [error, guardarError] = useState(false);

    // Funcicón que lee el presupuesto
    const definirPresupuesto = e => {
        guardarCantidad(parseInt(e.target.value, 10))
    }

    // Submit para definit el presupuesto
    const agregrarPresupuesto = e => {
        e.preventDefault();

        // Validar
        if (cantidad < 1 || isNaN(cantidad)) {
            guardarError(true);
            return;
        }

        // Si se pasa la validación
        guardarError(false);

    }

    return (
        <>
            <h2>Coloca tu presupuesto</h2>

            { error ? <Error mensaje="El Presupuesto es Incorrecto" /> : null}

            <form onSubmit={agregrarPresupuesto}>
                <input
                    type="number"
                    className="u-full-width"
                    placeholder="Coloca tu presupuesto"
                    onChange={definirPresupuesto}
                />

                <input
                    type="submit"
                    className="button-primary u-full-width"
                    value="Definir Presupuesto"
                />
            </form>
        </>
    );
}

export default Pregunta;