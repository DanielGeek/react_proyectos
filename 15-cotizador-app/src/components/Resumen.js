import React from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import { primeraLetrarMayuscula } from '../helper';


const ContenedorResumen = styled.div`
    padding: 1rem;
    text-align: center;
    background-color: #00838F;
    color: #FFF;
    margin-top: 1rem;
`;

export const Resumen = ({ datos }) => {

    // extraer de datos
    const { marca, year, plan } = datos;

    if (marca === '' || year === '' || plan === '') return null;

    return (
        <ContenedorResumen>
            <h2>Resumen de Cotización</h2>
            <ul>
                <li>Marca: {primeraLetrarMayuscula(marca)}</li>
                <li>Plan: {primeraLetrarMayuscula(plan)}</li>
                <li>Año del Auto: {year} </li>
            </ul>
        </ContenedorResumen>
    )
}

Resumen.propTypes = {
    datos: PropTypes.object.isRequired
}