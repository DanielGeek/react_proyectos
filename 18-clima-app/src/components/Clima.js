import React from 'react';
import PropTypes from 'prop-types';

export const Clima = ({ resultAPI }) => {

    // extraer los valores
    const { name, main } = resultAPI;

    if (!name) return null;

    // Grados Kelvin
    const kelvin = 273.15;

    return (
        <div className="card-panel white col s12">
            <div className="black-text">
                <h2>The weather of {name} is: </h2>
                <p className="temperatura">
                    {parseFloat(main.temp - kelvin, 10).toFixed(2)} <span> &#x2103; </span>
                </p>
                <p>Maximum temperature:
                    {parseFloat(main.temp_max - kelvin, 10).toFixed(2)} <span> &#x2103; </span>
                </p>
                <p>Minimum temperature:
                    {parseFloat(main.temp_min - kelvin, 10).toFixed(2)} <span> &#x2103; </span>
                </p>
            </div>
        </div>
    );
}

Clima.propTypes = {
    resultAPI: PropTypes.object.isRequired
}
