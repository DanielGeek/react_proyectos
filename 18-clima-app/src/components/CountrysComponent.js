import React from 'react';
import PropTypes from 'prop-types';

export const CountrysComponent = ({ countryAPI }) => {

    return (
        <option value={countryAPI.alpha2Code}>
            {countryAPI.name}
        </option>
    )
}

CountrysComponent.propTypes = {
    countryAPI: PropTypes.object.isRequired,
}

