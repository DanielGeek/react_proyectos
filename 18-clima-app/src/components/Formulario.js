import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Error } from './Error';
import { CountrysComponent } from './CountrysComponent';

export const Formulario = ({ search, setSearch, saveConsult, countrysAPI }) => {

    const [error, saveError] = useState(false);

    // extract city and country
    const { city, country } = search;

    // function put the elements in the state
    const handleChange = e => {
        setSearch({
            ...search,
            [e.target.name]: e.target.value
        });
    }
    // user submit form
    const handleSubmit = e => {
        e.preventDefault();

        // validator
        if (city.trim() === '' || country.trim() === ' ') {
            saveError(true);
            return;
        }

        saveError(false);

        saveConsult(true);
    }


    console.log(countrysAPI.length);

    useEffect(() => {
        // para que funcione el select con los paises obtenidos por api
        // imitialize dropdown
        var elems = document.getElementById("country");
        window.M.FormSelect.init(elems, {});
    }, []);

    return (
        <form
            onSubmit={handleSubmit}
        >
            {error ? <Error mensaje="All fields are required" /> : null}
            <div className="input-field col s12">
                <input
                    type="text"
                    name="city"
                    id="city"
                    value={city}
                    onChange={handleChange}
                />
                <label htmlFor="city">City: </label>
            </div>

            <div className="input-field col s12">
                <select
                    className="browser-default"
                    name="country"
                    id="country"
                    value={country}
                    onChange={handleChange}
                >

                    {
                        !countrysAPI.length ?
                            null
                            :
                            countrysAPI.map((countryAPI) => {
                                return <CountrysComponent key={countryAPI.alpha2Code} countryAPI={countryAPI} />
                            })
                    }

                    {/* <option value="">-- Select Country --</option>
                    <option value="AR">Argentina</option>
                    <option value="CL">Chile</option>
                    <option value="CO">Colombia</option>
                    <option value="CR">Costa Rica</option>
                    <option value="ES">España</option>
                    <option value="US">Estados Unidos</option>
                    <option value="MX">México</option>
                    <option value="PE">Perú</option>
                    <option value="VE">Venezuela</option> */}

                </select>
                {/* <label htmlFor="country">Country: </label> */}
            </div>
            <div className="input-field col s12">
                <button
                    type="submit"
                    className="waves-effect waves-light btn-large btn-block yellow accent-4 col s12"
                >Look for weather
                </button>
            </div>
        </form >
    )
}

Formulario.propTypes = {
    search: PropTypes.object.isRequired,
    setSearch: PropTypes.func.isRequired,
    saveConsult: PropTypes.func,
    countrysAPI: PropTypes.array
}