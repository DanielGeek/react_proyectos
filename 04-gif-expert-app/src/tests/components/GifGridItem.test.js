import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas den <GifGridItem />', () => {

    const title = 'Un tÃ­tutlo';
    const url = 'https://localhost/algo.jpg';

    test('debe de mostrar el componente correctamente ', () => {
        
        const wrapper = shallow( <GifGridItem title={ title } url={ url } />)
        expect( wrapper ).toMatchSnapshot();

    })
    
})
