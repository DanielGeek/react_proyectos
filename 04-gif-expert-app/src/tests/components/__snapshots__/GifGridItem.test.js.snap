// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas den <GifGridItem /> debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un títutlo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un títutlo
     
  </p>
</div>
`;
